droidlet/dialog/dialogue_task.py

```diff
@@ -0,0 +1,206 @@
+
+    """
+
+    def __init__(self, agent, task_data):
+        super().__init__(agent, task_data={})
+        response_options = task_data.get("response_options")
+        if not response_options:
+            raise ValueError("Cannot init a Say with no response options")
+
+        if type(response_options) is str:
+            self.response_options = [response_options]
+        else:
+            self.response_options = response_options
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+
+    @Task.step_wrapper
+    def step(self):
+        """Return one of the response_options."""
+        self.finished = True
+        self.agent.send_chat(random.choice(self.response_options))
+
+
+class BotCapabilities(Say):
+    """This class represents a sub-type of the Say Task to answer
+    something about the current capabilities of the bot, to the user.
+
+    """
+
+    def __init__(self, agent, task_data={}):
+        # Should put these in a config file somewhere...
+        response_options = [
+            'Try looking at something and tell me "go there"',
+            "Try asking to get something for you",
+            "Try asking me to dance",
+            "Try asking me to point at something",
+            "Try asking me to drop whatever is in my hand",
+        ]
+        super().__init__(agent, {"response_options": response_options})
+
+
+class ConfirmTask(Task):
+    """Confirm that a Task should be executed 
+
+    Args:
+        question: the question to ask the user
+        tasks: list of task objects
+        asked: flag to denote whether the clarification has been asked for
+
+    """
+
+    def __init__(self, agent, task_data={}):
+        task_data["blocking"] = True
+        super().__init__(agent, task_data=task_data)
+        self.question = task_data.get("question")  # chat text that will be sent to user
+        self.task_memids = task_data.get(
+            "task_memids"
+        )  # list of Task objects, will be pushed in order
+        self.asked = False
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+
+    @Task.step_wrapper
+    def step(self):
+        """Ask a confirmation question and wait for response."""
+        # Step 1: ask the question
+        if not self.asked:
+            task_list = [
+                Say(self.agent, {"response_options": self.question}),
+                AwaitResponse(self.agent, {"asker_memid": self.memid}),
+            ]
+            self.add_child_task(maybe_task_list_to_control_block(task_list, self.agent))
+            self.asked = True
+            return
+        # Step 2: check the response and add the task if necessary
+        self.finished = True
+        # FIXME: change this to sqly when syntax for obj searches is settled:
+        t = self.agent.memory.get_triples(subj=self.memid, pred_text="dialogue_task_response")
+        chat_mems = [self.agent.memory.get_mem_by_id(triples[2]) for triples in t]
```

Comment:
For better understandability, can we add a comment about the format of each triple here ?
`(subj, pred, obj)`